src/main/java/io/vertx/ext/hawkular/impl/VertxMetricsImpl.java

```diff
@@ -121,12 +122,18 @@ public void eventBusInitialized(EventBus bus) {
     Context context = vertx.getOrCreateContext();
     sender = new Sender(vertx, options, context);
     scheduler = new Scheduler(vertx, options, context, sender);
-    scheduler.register(httpServerMetricsSupplier);
-    scheduler.register(httpClientMetricsSupplier);
-    scheduler.register(netServerMetricsSupplier);
-    scheduler.register(netClientMetricsSupplier);
-    scheduler.register(datagramSocketMetricsSupplier);
-    scheduler.register(eventBusMetrics);
+    if(this.options.isMetricsTypeDisabled(MetricsType.HTTP_SERVER))
+      scheduler.register(httpServerMetricsSupplier);
```

Comment:
I believe we actually want the opposite: if the type is disabled, don't register the supplier.